### ISSUE TITLE
Section mismatch in reference from the function __trace_early_add_events() to the function .init.text:__trace_early_add_new_event()

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#1174]
```sh
$ ARCH=s390 CROSS_COMPILE=s390x-linux-gnu- make -j72 CC=clang defconfig vmlinux
...
WARNING: modpost: vmlinux.o(.text+0x225234): Section mismatch in reference from the
function __trace_early_add_events() to the function .init.text:__trace_early_add_new_event()
```
likely `__trace_early_add_events` is not marked `init` (and should be) yet calls `__trace_early_add_new_event` which is marked `init`. Or `init` should be removed from `__trace_early_add_new_event`

**Comment by nickdesaulniers** [comment]
I'm seeing this for 32b arm as well, so maybe a recent regression upstream.

**Comment by nickdesaulniers** [comment]
Unreproducible; I suspect this was fixed by ce66f6136460a51acfc32de4481fe8fd69dfd50b in v5.10-rc1.
